src/components/Stats.tsx

```diff
@@ -1,118 +1,100 @@
+  );
 
-    return (
-      <>
-        {libraryStats && (
-          <LibraryStats library={libraryData} stats={libraryStats} />
-        )}
-        {totalStats && <LibraryStats stats={totalStats} />}
-        {this.props.fetchError && (
-          <ErrorMessage error={this.props.fetchError} />
-        )}
+  return (
+    <>
+      {targetLibraryData && (
+        <LibraryStats library={targetLibraryKey} stats={targetLibraryData} />
+      )}
+      {!targetLibraryData && summaryStatistics && (
+        <LibraryStats stats={summaryStatistics} />
+      )}
+      {fetchError && <ErrorMessage error={fetchError} />}
+      {!isLoaded && <LoadingIndicator />}
+    </>
+  );
+};
 
-        {!this.props.isLoaded && <LoadingIndicator />}
-      </>
-    );
+/**
+ * Convert the fetched data from the sparse API format.
+ * - Adds a list of collections to each library, one for each collectionId.
+ * - Adds summary statistics in the form of a LibraryStatistics object.
+ * @param {StatisticsData} statistics - Statistics data fetched via API.
+ */
+export const normalizeStatistics = (statistics): StatisticsData => {
+  if (!statistics) {
+    return statistics;
   }
 
-  UNSAFE_componentWillMount() {
-    if (this.props.fetchStats) {
-      this.props.fetchStats();
-    }
-    if (this.props.fetchLibraries()) {
-      this.props.fetchLibraries();
-    }
-  }
-}
+  const collectionIdMap = Object.assign(
+    {},
+    ...statistics.collections.map((c) => ({ [c.id]: c }))
+  );
+  const libraries = statistics.libraries.map((l) => ({
+    ...l,
+    collections: l.collectionIds.map((id) => collectionIdMap[id]),
+  }));
+  const collectionIds = statistics.collections.map((c) => c.id);
 
-function mapStateToProps(state, ownProps) {
-  return {
-    stats: state.editor.stats.data,
-    libraries:
-      state.editor.libraries &&
-      state.editor.libraries.data &&
-      state.editor.libraries.data.libraries,
-    fetchError: state.editor.stats.fetchError,
-    isLoaded: state.editor.stats.isLoaded,
+  const summaryStatistics: LibraryStatistics = {
+    key: "_summary_",
+    name: "Summary Statistics",
+    patronStatistics: statistics.patronSummary,
+    inventorySummary: statistics.inventorySummary,
+    collectionIds,
+    collections: statistics.collections,
   };
-}
 
-function mapDispatchToProps(dispatch) {
-  const actions = new ActionCreator();
-  return {
-    fetchStats: () => dispatch(actions.fetchStats()),
-    fetchLibraries: () => dispatch(actions.fetchLibraries()),
-  };
-}
+  return { ...statistics, libraries, summaryStatistics };
+};
+
+const useStatistics = (): StatsLocalState => {
+  const data = useSelector(statsStateSelector.data);
+  const isLoaded = useSelector(statsStateSelector.isLoaded);
+  const fetchError = useSelector(statsStateSelector.fetchError);
+
+  const dispatch = useDispatch();
+  useEffect(() => {
+    const actions = new ActionCreator();
+    dispatch(actions.fetchStatistics());
+  }, []);
```

Comment:
... and picks up the fetch previously performed in the `componentWillMount` lifecycle method.